internal/telemetry/tracker.go

```diff
@@ -63,6 +67,7 @@ func newTracker(ctx context.Context) (*tracker, error) {
 		cacheDir:         cacheDir,
 		store:            telemetryStore,
 		storeSet:         storeSet,
+		maxBatchSize:     defaultMaxBatchSize,
```

Comment:
Personally I think this is getting too complex, I think we should have either number of items in cache OR max cache file size and always send everything.

Reply:
+1 on the suggestion here. Limiting to a max number of items makes the performance predictable and also achieves the key product goal of tracking the initial events as a priority.

Reply:
Sorry for the misunderstanding - perhaps I can think up better variable names to make it clearer...

`maxCacheFileSize` is the maximum size of the cache file.
`maxBatchSize` is the maximum number of events that we'll send in a POST request.

I could rename `maxBatchSize` to `maxRequestPayloadSize` for example.

Reply:
I understand. That will indeed simplify things, though we will sacrifice caching a large number of events, ie. it would be unreasonable to cache 20 thousand events in a 10MB cache file and send all those events within one command run.

So, I have made changes to specify only the maximum cache file size, choosing a value of 500KB. That should allow us to cache approximately 800 events, while ensuring that our POST request payload also won't exceed 500KB.

I think that's a reasonable size limit - let me know what you think.

Reply:
I think 800 events are ok, I think it is unlikely users will ever run a situation where they ran so many commands without a profile but that is possible.